src/io/jesi/backpack/macros.cljc

```diff
@@ -3,38 +3,46 @@
   #?(:cljs (:require-macros [io.jesi.backpack.macros]))
   (:require
     [clojure.core]
-    [io.jesi.backpack.fn :refer [noop]]
-    [io.jesi.backpack.miscellaneous :refer [cljs-env?]])
+    [io.jesi.backpack.env :as env]
+    [io.jesi.backpack.fn :refer [noop]])
   #?(:cljs (:require [cljs.core :refer [IFn]]))
   #?(:clj
      (:import
        (clojure.lang IFn))))
 
-(defmacro import-vars
-  [& imports]
+(defmacro import-vars [& imports]
   `(do
      ~@(apply concat
-         (for [[ns & names] imports
-               name names
-               :let [src (symbol (str ns) (str name))
-                     meta (meta (resolve src))
-                     arglists (get meta :arglists)
-                     doc (get meta :doc "")]]
-           `(
-             (def ~name ~doc ~src)
-             (alter-meta! #'~name assoc :arglists '~arglists))))))
-
-(defmacro if-cljs
-  "Return `then` if we are generating cljs code, and `else` for Clojure code"
-  [then else]
-  (if (cljs-env? &env)
-    then
-    else))
+         (for [import imports
+               :let [vars (if (symbol? import)
+                            (do
+                              (require import)
+                              (vals (ns-publics import)))
+                            (let [ns (first import)]
+                              (map
+                                (fn [name]
+                                  (require ns)
+                                  (let [sym (symbol (str ns) (str name))
+                                        var (resolve sym)]
+                                    (when (nil? var)
+                                      (throw (ex-info (str "Could not resolve var " sym) {:symbol sym
+                                                                                          :ns     *ns*
+                                                                                          :env    &env})))
```

Comment:
Need to resolve the symbol to get the meta. So now we error if couldn't resolve sym